build_tools/check_backticks.py

```diff
@@ -0,0 +1,121 @@
+            # constant's value is a string
+            # the node represents a docstring
+            # See https://docs.python.org/3/library/ast.html#abstract-grammar
+            docstring = node.value.value
+            lineno = node.value.lineno
+            docstrings[lineno] = docstring
+
+    return docstrings
+
+
+def find_invalid_backtick_text(docstring):
+    """Find invalid backtick text in a docstring."""
+    # remove all the code blocks to avoid interference
+    # they are allowed to use backticks in any way
+    docstring = re.sub(r"```.*?```", "", docstring, flags=re.DOTALL)
+
+    # remove all double-backticks to avoid interference
+    # we are looking only for invalid single-backtick
+    docstring = re.sub(r"``.*?``", "", docstring)
+
+    all_backtick_text = re.findall(r"`.*?`", docstring)
+    # expressions like :math:`d(x, y):= (x-y)^2` are valid cases
+    valid_backtick_text = re.findall(r":.*?:(`.*?`)", docstring)
+
+    # find all the invalid backtick code snippets
+    invalid_backtick_text = set()
+    for text in all_backtick_text:
+        if text in valid_backtick_text:
+            continue
+        invalid_backtick_text.add(text)
+
+    return invalid_backtick_text
+
+
+def main():
+    """Execute the main function of the script."""
+    # parse command line arguments
+    parser = argparse.ArgumentParser(
+        description="Test script to check for invalid use of single-backticks."
+    )
+    parser.add_argument(
+        "folder_path",
+        nargs="?",
+        default="./sktime",
+        help="Folder path to search for Python files",
+    )
+    args = parser.parse_args()
+
+    folder_path = args.folder_path
+    results = {}
+
+    # list all the python files in the project
+    py_files = find_py_files(folder_path)
+
+    for file in py_files:
+        docstrings = extract_docstrings(file)
+        results_on_file = {}
+
+        for lineno, docstring in docstrings.items():
+            invalid_backtick_text = find_invalid_backtick_text(docstring)
+
+            if len(invalid_backtick_text) > 0:
+                results_on_file[lineno] = invalid_backtick_text
+
+        if len(results_on_file) > 0:
+            results[file] = results_on_file
+
+    # print the lines along with the invalid backticks text
+    print(f'Results in "{folder_path}"')  # noqa: T201
+    if len(results) > 0:
+        print(f"Total Files with invalid backticks: {len(results)}")  # noqa: T201
+        for filename, result in results.items():
+            for lineno, errors in result.items():
+                print(f"{filename}:{lineno} {' '.join(errors)}")  # noqa: T201
+    else:
+        print("No invalid backticks found")  # noqa: T201
+
+
+if __name__ == "__main__":
+    """
+    Usage: defaults to "./sktime"
+    python build_tools/check_backticks.py
+
+    Usage: folder path as argument
+    python build_tools/check_backticks.py sktime/classification/distance_based
+    """
```

Comment:
If you want to include a custom usage documentation, `usage` argument of `ArgumentParser` is probably the better place.

Reply:
I added this just for someone looking into the file, it was not really meant to be usage documentation, but if you say, I can put it there.